pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -223,26 +161,30 @@ func (r Requirements) Validate() (errs error) {
 			r.hasRequirement(withKeyAndOperator(requirement.Key, v1.NodeSelectorOpExists))) {
 			errs = multierr.Append(errs, fmt.Errorf("operator %s cannot coexist with other operators for key %s", v1.NodeSelectorOpDoesNotExist, requirement.Key))
 		}
+		// Excludes cases when In and NotIn have overlaps.
+		if requirement.Operator == v1.NodeSelectorOpIn && r.hasRequirement(withKeyOperatorAndValues(requirement.Key, v1.NodeSelectorOpNotIn, requirement.Values)) {
+			errs = multierr.Append(errs, fmt.Errorf("operators %s and %s have common values for key %s", v1.NodeSelectorOpIn, v1.NodeSelectorOpNotIn, requirement.Key))
+		}
 	}
 	for key := range r.Keys() {
-		if r.Get(key).Len() == 0 {
+		if r.Get(key).Len() == 0 && !r.hasRequirement(withKeyAndOperator(key, v1.NodeSelectorOpDoesNotExist)) {
 			errs = multierr.Append(errs, fmt.Errorf("no feasible value for key %s", key))
 		}
 	}
 	return errs
 }
 
-// Compatible ensures the provided requirements can be met. It is
-// non-commutative (i.e., A.Compatible(B) != B.Compatible(A))
+// Compatible ensures the provided requirements can be met.
 //gocyclo:ignore
 func (r Requirements) Compatible(requirements Requirements) (errs error) {
 	for _, key := range r.Keys().Union(requirements.Keys()).UnsortedList() {
 		// Key must be defined if required
 		if values := requirements.Get(key); values.Len() != 0 && !values.IsComplement() && !r.hasRequirement(withKey(key)) {
 			errs = multierr.Append(errs, fmt.Errorf("require values for key %s but is not defined", key))
 		}
-		// Values must overlap
-		if values := r.Get(key); values.Intersection(requirements.Get(key)).Len() == 0 {
+		// Values must overlap except DoesNotExist operator
+		// Excluding DoesNotExist cases to avoid generating mutiple error mesasges for the same error.
```

Comment:
Can you show a case of what this error message looks like? Line 187 makes my brain hurt.
```suggestion
		// Excluding DoesNotExist cases to avoid generating multiple error messages for the same error.
```

Reply:
DoesNotExist requires DoesNotExist, key topology.kubernetes.io/zone; Exists prohibits DoesNotExist, key topology.kubernetes.io/zone